bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/client/ConsumedThingImpl.java

```diff
@@ -0,0 +1,246 @@
+     */
+    ConsumedThingImpl(WebSocketClient webSocketClient, HttpClient httpClient, URI webThingURI,
+            ScheduledExecutorService executor, Consumer<String> errorHandler) throws IOException {
+        this(httpClient, webThingURI, executor, errorHandler, WebSocketConnectionFactory.instance(webSocketClient));
+    }
+
+    /**
+     * constructor
+     *
+     * @param httpClient the http client to use
+     * @param webthingUrl the identifier of a WebThing resource
+     * @param executor executor to use
+     * @param errorHandler the error handler
+     * @param webSocketConnectionFactory the Websocket connectino fctory to be used
+     * @throws IOException if the WebThing can not be connected
+     */
+    ConsumedThingImpl(HttpClient httpClient, URI webthingUrl, ScheduledExecutorService executor,
+            Consumer<String> errorHandler, WebSocketConnectionFactory webSocketConnectionFactory) throws IOException {
+        this(httpClient, webthingUrl, executor, errorHandler, webSocketConnectionFactory, DEFAULT_PING_PERIOD);
+    }
+
+    /**
+     * constructor
+     *
+     * @param httpClient the http client to use
+     * @param webthingUrl the identifier of a WebThing resource
+     * @param executor executor to use
+     * @param errorHandler the error handler
+     * @param webSocketConnectionFactory the Websocket connectino fctory to be used
+     * @param pingPeriod the ping period tothe the healthiness of the connection
+     * @throws IOException if the WebThing can not be connected
+     */
+    ConsumedThingImpl(HttpClient httpClient, URI webthingUrl, ScheduledExecutorService executor,
+            Consumer<String> errorHandler, WebSocketConnectionFactory webSocketConnectionFactory, Duration pingPeriod)
+            throws IOException {
+        this.webThingURI = webthingUrl;
+        this.httpClient = httpClient;
+        this.errorHandler = errorHandler;
+        this.description = new DescriptionLoader(httpClient).loadWebthingDescription(webThingURI,
+                Duration.ofSeconds(20));
+
+        // opens a websocket downstream to be notified if a property value will be changed
+        this.websocketDownstream = webSocketConnectionFactory.create(this.getEventStreamUri(), executor, this::onError,
+                pingPeriod);
+    }
+
+    private URI getPropertyUri(String propertyName) {
+        var optionalProperty = description.getProperty(propertyName);
+        if (optionalProperty.isPresent()) {
+            var propertyDescription = optionalProperty.get();
+            for (var link : propertyDescription.links) {
+                if ((link.rel != null) && (link.href != null) && link.rel.equals("property")) {
+                    return webThingURI.resolve(link.href);
+                }
+            }
+        }
+        throw new RuntimeException(
+                "WebThing " + webThingURI + " does not support a property uri. WebThing description: " + description);
```

Comment:
Are you aware that your binding will crash if this exception is raised? If this can only happen due to a bug in your code, I'd make this rather an `IllegalStateException`. If this is expected to happen, it's better to make it a checked (custom) exception and handle it in the calling method. Same for below.

Reply:
done